libs/feature/search/src/lib/results-table/results-table.component.spec.ts

```diff
@@ -0,0 +1,209 @@
+        ])
+        expect(await firstValueFrom(component.isSortedBy('title'))).toBe('desc')
+        expect(await firstValueFrom(component.isSortedBy('score'))).toBe('asc')
+        expect(await firstValueFrom(component.isSortedBy('owner'))).toBe(null)
+      })
+    })
+  })
+
+  describe('selection', () => {
+    beforeEach(() => {
+      searchFacade.results$.next([
+        {
+          uniqueIdentifier: '1',
+        },
+        {
+          uniqueIdentifier: '2',
+        },
+        {
+          uniqueIdentifier: '3',
+        },
+      ] as any)
+    })
+
+    describe('#isChecked', () => {
+      it('should return true when the record is in the selectedRecords array', async () => {
+        selectionService.selectedRecordsIdentifiers$.next(['1', '2'])
+        const record = { uniqueIdentifier: '2' } as CatalogRecord
+        expect(await firstValueFrom(component.isChecked(record))).toBe(true)
+      })
+
+      it('should return false when the record is not in the selectedRecords array', async () => {
+        selectionService.selectedRecordsIdentifiers$.next(['1', '2', '3'])
+        const record = { uniqueIdentifier: '4' } as CatalogRecord
+        expect(await firstValueFrom(component.isChecked(record))).toBe(false)
+      })
+    })
+
+    describe('#handleRecordSelectedChange', () => {
+      it('should call selectRecords when checkbox is clicked', () => {
+        const record = { uniqueIdentifier: '1' }
+        component.handleRecordSelectedChange(true, record as CatalogRecord)
+        expect(selectionService.selectRecords).toHaveBeenCalledWith([record])
+      })
+    })
+
+    describe('#isAllSelected', () => {
+      it('returns true if all records are selected', async () => {
+        selectionService.selectedRecordsIdentifiers$.next([
+          '1',
+          '2',
+          '3',
+          '4',
+          '5',
+        ])
+        expect(await firstValueFrom(component.isAllSelected())).toBe(true)
+      })
+      it('returns false otherwise', async () => {
+        selectionService.selectedRecordsIdentifiers$.next(['1'])
+        expect(await firstValueFrom(component.isAllSelected())).toBe(false)
+      })
+    })
+
+    describe('#isSomeSelected', () => {
+      it('returns false if all records are selected', async () => {
+        selectionService.selectedRecordsIdentifiers$.next([
+          '1',
+          '2',
+          '3',
+          '4',
+          '5',
+        ])
+        expect(await firstValueFrom(component.isSomeSelected())).toBe(false)
+      })
+      it('returns true if more than one record selected', async () => {
+        selectionService.selectedRecordsIdentifiers$.next(['2', '3'])
+        expect(await firstValueFrom(component.isSomeSelected())).toBe(true)
+      })
+      it('returns false if no record selected', async () => {
+        selectionService.selectedRecordsIdentifiers$.next(['4', '5'])
+        expect(await firstValueFrom(component.isSomeSelected())).toBe(false)
+      })
```

Comment:
I don't understand this test compared to the previous one? Are there no selected records because 4 and 5 are not part of the possible selection? If so, this should be explicit in the description.